drivers/utils.go

```diff
@@ -0,0 +1,34 @@
+package drivers
+
+import (
+	"database/sql"
+	"errors"
+
+	"github.com/jorgerojas26/lazysql/helpers/logger"
+	"github.com/jorgerojas26/lazysql/models"
+)
+
+func queriesInTransaction(db *sql.DB, queries []models.Query) (err error) {
+	trx, err := db.Begin()
+	if err != nil {
+		return err
+	}
+	defer func() {
+		rErr := trx.Rollback()
+		// sql.ErrTxDone is returned when trx.Commit was already called
+		if !errors.Is(rErr, sql.ErrTxDone) {
+			err = errors.Join(err, rErr)
+		}
+	}()
+
+	for _, query := range queries {
+		logger.Info(query.Query, map[string]any{"args": query.Args})
+		if _, err := trx.Exec(query.Query, query.Args...); err != nil {
+			return err
+		}
+	}
+	if err := trx.Commit(); err != nil {
+		return err
+	}
```

Comment:
I might be wrong, bit I don't think it's working, because of the scope of variable in the if



When you use `if err := whatever(); err != nil`, you introduce a new variable with a very limited scope, so the err here, is not the one the defer will have access to. For me, the code would behave as if you used `if errWhatever := whatever(); errWhatever != nil`

I would suggest using this

```suggestion
	for _, query := range queries {
		logger.Info(query.Query, map[string]any{"args": query.Args})
		_, err = trx.Exec(query.Query, query.Args...); 
		if err != nil {
			return err
		}
	}
	err = trx.Commit(); 
	if err != nil {
		return err
	}
```

Reply:
Yeah, this is kinda confusing and that's why I'm trying to avoid named returns.

https://go.dev/play/p/5OSJ8UBxnj1

It works, because the `(err error)` is what was returned, not what was assigned to the variable in the function body.
Here it may be more clear: https://go.dev/play/p/gdrDvaaA0Hj

Reply:
Thanks for checking.

I do hate named return too :grimacing: 

I thought the defer would cause issue, but yes it's a pain it requires to use a named parameter (it doesn't work without)

Another solution to write this

```go
	trx, err := db.Begin()
	if err != nil {
		return err
	}
	err = func() error {
		for _, query := range queries {
		logger.Info(query.Query, map[string]any{"args": query.Args})
		if _, err := trx.Exec(query.Query, query.Args...); err != nil {
			return err
		}
		return nil
	}()
	if err != nil {
		return errors.Join(err, rx.Rollback())
	}

	if err := trx.Commit(); err != nil {
		return err
	}
```